syntax = "proto3";

package datanode;

// Mensajes de solicitud y respuesta actualizados.
message ReplicateDataRequest {
  string file_name = 1;
  bytes data = 2;
}

message ReplicateDataResponse {
  bool success = 1;
  string error_message = 2;
}

message NotifyNameNodeRequest {
  string file_name = 1;
  int32 replicas = 2;
}

message NotifyNameNodeResponse {
  bool success = 1;
  string error_message = 2;
}

message ReadFileRequest {
  string file_name = 1;
}

message ReadFileResponse {
  bytes data = 1;
  bool success = 2;
  string error_message = 3;
}

// Servicio DataNode con las operaciones soportadas.
service DataNodeService {
  rpc ReplicateData(ReplicateDataRequest) returns (ReplicateDataResponse);
  rpc NotifyNameNode(NotifyNameNodeRequest) returns (NotifyNameNodeResponse);
  rpc ReadFile(ReadFileRequest) returns (ReadFileResponse);
}
